lib/ansible/module_utils/basic.py

```diff
@@ -411,9 +411,8 @@ def remove_values(value, no_log_strings):
         old_data, new_data = deferred_removals.popleft()
         if isinstance(new_data, Mapping):
             for old_key, old_elem in old_data.items():
-                new_key = _remove_values_conditions(old_key, no_log_strings, deferred_removals)
```

Comment:
This seems like the heart of something unmentioned in the changelogs but it doesn't seem okay to do... at least, not without a deprecation period.  Changing keys by default isn't without its drawbacks but anyone who depends upon it to keep secure values hidden will run into problems with this.

I think I heard something about the problem being with specific keys which should not be removed (like `changed` or `failed` ?).  If so, those could go into an always_allow list of some sort.  Another alternative which is something of a compromise might be to allow toplevel keys to be output unchanged but to continue to remove values on any keys inside of toplevel containers.

Reply:
This change was new to 2.10, and after discussing it with prodsec, and the core eng team, we decided to revert this general change, and push the change onto module authors, thus handling it in `uri` specifically at this time.

Reply:
@abadger This comes from a change that was merged for 2.10 so it's not yet released. We are basically reverting most of 69653 (I'm keeping the test that was added though) and replacing with this code that is specific for the uri module. You are right and folks are complaining that we should not modify the keys with such a big hammer (see issue #70455).

Reply:
<nod> yeah.  I thought I had added it back in early 2.x but my memory was faulty.  Objection here withdrawn :-)